### ISSUE TITLE
Move Arc App's brexit() method to LocoKit

### ISSUE THREAD

**Issue by sobri909** [sobri909/LocoKit#13]
Splitting this task off from #11, because it's not a requirement for that task, but will be nice to eventually have in LocoKit instead of hidden away in Arc App.

## Background

The `brexit()` method allows for extracting an `ItemSegment` out of a `TimelineItem` into a separate item (for example a brief visit to a specific mall shop within a larger mall visit) and intelligently maintaining the timeline's linked list as well as extracting path items between the new items, as appropriate.

The reason the method is called "brexit" is because if the post processing doesn't go how you want it, you can get left with an unholy mess and wish you never tried in the first place.

**Comment by sobri909** [comment]
This is in the `develop` branch now, although with a couple of vital bits of functionality still to come.

*[**Aside:** I mistagged the issue number on the initial commits, tagging them as #11 instead of #13. Oops.]*

The new method can be found in the new [PersistentProcessor](https://github.com/sobri909/LocoKit/blob/develop/LocoKit/LocalStore/PersistentProcessor.swift), and used thus:

```swift
PersistentProcessor.extractItem(for: itemSegment, in: store) { newItem in
    // do something with the newly created TimelineItem
}
```

It takes an `ItemSegment` (ie a contiguous collection of `LocomotionSample`), and creates a new `TimelineItem` from that segment, by removing those samples from their current item, and inserting the new item into the timeline. 

The new item will be a `PersistentVisit` if the segment's activity type is `.stationary`, otherwise it will be a `PersistentPath`.

Basically this lets you take an item segment from a timeline item's `item.segments` and extract it out into a separate timeline item, which will then be sensibly connected up to the timeline sequence.

The main use cases being things like extracting brief visits out of paths (for example a brief stop at the traffic lights while driving), extracting smaller visits out of larger visits (for example a brief visit to a shop inside a mall), etc. 

You can see this in action in Arc App's "Recorded Activities" view on timeline items. I'll post a screenshot shortly.

**Comment by sobri909** [comment]
So here's an example of a `Path` with an `ItemSegment` of `.walking` type, which could be extracted with the new method. 

The screenshot shows a view of a single `Path` item which is comprised or two `ItemSegments`, one of which is cycling and the other walking. 

The `TimelineProcessor` merged those two segments together into a single Path item because the walking segment wasn't long enough to justify standing alone. But the new method allows you to overrule that decision, and extract the segment out into its own timeline item. 

This is what Arc App does when you tap on those rows in a timeline item's "Recorded Activities" view. 

![ItemSegments](https://user-images.githubusercontent.com/1203835/40649305-c48b9004-635a-11e8-96bf-99d948b87acb.jpg)
